segmentation/utils/spatial_analysis_utils.py

```diff
@@ -253,3 +253,48 @@ def calculate_enrichment_stats(close_num, close_num_rand):
     dims = ["stats", "rows", "cols"]
     stats_xr = xr.DataArray(stats_data, coords=coords, dims=dims)
     return stats_xr
+
+
+def compute_neighbor_counts(fov_data, dist_matrix, distlim, self_neighbor=True,
+                            cell_label_col="cellLabelInImage"):
+    """Calculates the number of neighbor phenotypes for each cell. The cell counts itself as a neighbor.
```

Comment:
We don't need to state that the cell counts itself as a neighbor in the main summary docstring if it's specified with self_neighbor argument description